cli/image_prune.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alibaba/pouch/apis/filters"
+
+	"github.com/spf13/cobra"
+)
+
+// imagePruneDescription is used to remove all unused images.
+var imagePruneDescription = "Remove all unused images"
+
+// ImagePruneCommand use to remove all stopped images.
+type ImagePruneCommand struct {
+	baseCommand
+
+	force      bool
+	flagFilter []string
+}
+
+// Init initialize "image prune" command.
+func (i *ImagePruneCommand) Init(c *Cli) {
+	i.cli = c
+	i.cmd = &cobra.Command{
+		Use:   "prune [OPTIONS]",
+		Short: "Remove all unused images",
+		Long:  imagePruneDescription,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return i.runImagePrune(args)
+		},
+		Example: i.example(),
+	}
+	i.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (i *ImagePruneCommand) addFlags() {
+	flagSet := i.cmd.Flags()
+
+	flagSet.BoolVarP(&i.force, "force", "f", false, "Do not prompt for confirmation")
+	flagSet.StringSliceVarP(&i.flagFilter, "filter", "", []string{}, "Filter output based on conditions provided, filter support reference, since, before")
+}
+
+// runImagePrune is used to remove unused images.
+func (i *ImagePruneCommand) runImagePrune(args []string) error {
+	ctx := context.Background()
+	apiClient := i.cli.Client()
+
+	imageFilterArgs, err := filters.FromFilterOpts(i.flagFilter)
+
+	if err != nil {
+		return err
+	}
+
+	if !i.force {
+		fmt.Println("WARNING! This will remove all dangling images..")
+		fmt.Print("Are you sure you want to continue? [y/N]")
+		var input string
+		fmt.Scanf("%s", &input)
+		if input != "Y" && input != "y" {
+			fmt.Println("Total reclaimed space: 0B")
+			return nil
+		}
+	}
+
+	resp, err := apiClient.ImagePrune(ctx, imageFilterArgs)
+	if err != nil {
+		return err
+	}
+
+	for _, imgPruneInfo := range resp.ImagesDeleted {
+		fmt.Println("untagged:")
+		fmt.Print(imgPruneInfo.Untagged)
+		fmt.Println("Deleted Images:")
+		fmt.Print(imgPruneInfo.Deleted)
+	}
+
+	fmt.Printf("\nTotal reclaimed space: %d B\n", resp.SpaceReclaimed)
```

Comment:
It is better to use this function to make reclaimed size more readable.
https://github.com/alibaba/pouch/blob/7235f82ae43083197687c714e68bdf1ab425b22c/pkg/utils/utils.go#L27